cedar-policy/src/api.rs

```diff
@@ -1428,6 +1428,79 @@ impl Schema {
     pub fn action_entities(&self) -> Result<Entities, EntitiesError> {
         Ok(Entities(self.0.action_entities()?))
     }
+
+    /// Returns an iterator over every entity type that can be a principal for any action in this schema
+    ///
+    /// Note: this iterator may contain duplicates.
+    pub fn principals(&self) -> impl Iterator<Item = &EntityTypeName> {
```

Comment:
I feel like a lot of use cases for this function won't want duplicates, so we should make this easier by adding a doc-test showing this function used and collected into a hashset.